sys/include/event.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_event Event Queue
+ * @ingroup     sys
+ * @brief       Provides an Event loop
+ *
+ * This module offers an event queue framework like libevent or libuev.
+ *
+ * An event queue is basically a FIFO queue of events, with some functions to
+ * efficiently and safely handle adding and getting events to / from such a
+ * queue.
+ * An event queue is bound to a thread, but any thread or ISR can put events
+ * into a queue.
+ *
+ * An event is a structure containing a pointer to an event handler. It can be
+ * extended to provide context or arguments to the handler.  It can also be
+ * embedded into existing structures (see examples).
+ *
+ * Compared to msg or mbox, this some fundamental differences:
+ *
+ * 1. events are "sender allocated". Unlike msg_send(), event_post() never
+ *    blocks or fails.
+ * 2. events contain everything necessary to handle them, thus a thread
+ *    processing the events of an event queue doesn't need to be changed in
+ *    order to support new event types.
+ * 3. events can be safely used (and actually perform best) when used within
+ *    one thread, e.g., in order to create a state-machine like process flow.
+ *    This is not (easily) possible using msg queues, as they might fill up.
+ * 4. an event can only be queued in one event queue at the same time.
+ *    Notifying many queues using only one event object is not possible with
+ *    this imlementation.
+ *
+ * At the core, event_wait() uses thread flags to implement waiting for events
+ * to be queued. Thus event queues can be used safely and efficiently in combination
+ * with thread flags and msg queues.
+ *
+ * Examples:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * // simple event handler
+ * static void handler(event_t *event)
+ * {
+ *    printf("triggered 0x%08x\n", (unsigned)event);
+ * }
+ *
+ * static event_t event = { .handler=handler };
```

Comment:
Sorry, only noticed this during reading the parsed doc: spaces around inner `=`

Reply:
done